cluster.go

```diff
@@ -1003,6 +1004,24 @@ func (c *Cluster) StateSync(ctx context.Context) error {
 			continue
 		}
 
+		var t time.Time
+		if !currentPin.PinOptions.ExpireAt.Equal(t) && currentPin.PinOptions.ExpireAt.Before(time.Now()) {
```

Comment:
don't call time.Now() for every pin.

Reply:
This is shaky. Better to define a ZeroTime constant in types.h (per Protobufs definition it is the result of : `time.Unix(0, 0).UTC()` (https://github.com/golang/protobuf/blob/master/ptypes/timestamp.go#L95) (or better, the result of doing pbuf.Timestamp(nil)) and compare on that.

Reply:
There's also an IsZero().

Reply:
Yes, but `time.Unix(0,0)` is not zero
An uninitialized time variable is zero.

Even though, I didn't anything like you can't manually get zero time, I have inferred it so.
The function `t.UnixNano()`  (say we have `var t time.Time`) results in undefined for before year 1678 or after year 2262 (limitations of int64). So naturally you can't get back zero time using `time.Unix(sec int64, nsec int64)` 

I think it can be achieved by using `Sub(d duration)`, but all that pain doesn't seem worth taking.

Reply:
Concentrate on the point of checking that the "unset" time you get out of the protobuf can be identified as an unset time. If IsZero()  != time.Unix(0,0), then the line above is buggy which is exactly why I asked to define ZeroTime exactly as what protobuf considers as a zero/unset time.